src/component/ResponsiveContainer.tsx

```diff
@@ -83,7 +82,16 @@ export const ResponsiveContainer = forwardRef(
         containerHeight: containerRef.current.clientHeight,
       };
     }, []);
+    const getDataProps = (props: Partial<Props>) => {
```

Comment:
This does seem unrelated. Is it? 🤔

Reply:
Yes and no - there was no great way to get the responsive container reference without adding a testid (although className would've worked, id would've worked).

We have already proposed doing something like this elsewhere https://github.com/recharts/recharts/issues/3253#issuecomment-1423448329 and I think this is a great way of allowing users to specify data-* props they want on base elements without it messing with our implementation or being super specific.

I did ask this to be refactored to be elsewhere, maybe ideally we make the tests pass with `className` and then in a separate PR we add this helper function in a util file and add this ability back to responsive container

Reply:
@branberry what do you think?

Reply:
@ckifer If you'd like, I could move this function out into a separate file as a part of another PR so that we aren't blocking this one. Does that work for you?

Reply:
I'd rather remove this from this PR and add it back in it's own PR entirely rather than hiding the initial addition here

Reply:
I wonder whether we should even land this separately on master.

Reply:
Snapshots will succeed when we get this into master - I think our CI is running on master always as compared to on the target branch